Rules/UseUsingScopeModifierInNewRunspaces.cs

```diff
@@ -0,0 +1,430 @@
+                    startLineNumber: ast.Extent.StartLineNumber,
+                    endLineNumber: ast.Extent.EndLineNumber,
+                    startColumnNumber: ast.Extent.StartColumnNumber,
+                    endColumnNumber: ast.Extent.EndColumnNumber,
+                    text: varWithUsing,
+                    file: ast.Extent.File,
+                    description: description
+                )
+            };
+        }
+
+        /// <summary>
+        /// GetSessionName: Retrieves the name of the session (that Invoke-Command is run with).
+        /// </summary>
+        /// <param name="commandAst"></param>
+        private static string GetSessionName(CommandAst commandAst)
+        {
+            if (!(commandAst.CommandElements.First<Ast>(
+                e => e is CommandParameterAst parameterAst &&
+                     parameterAst.ParameterName.Equals(
+                         "session", StringComparison.OrdinalIgnoreCase)) is CommandParameterAst sessionParameterAst))
+            {
+                return "";
+            }
+
+            var sessionParamAstIndex = commandAst.CommandElements.IndexOf(sessionParameterAst);
+
+            return commandAst
+                .CommandElements[sessionParamAstIndex + 1]
+                .Extent
+                .Text
+                .Trim();
+        }
+
+        /// <summary>
+        /// GetAssignedVarsInSession: Retrieves all previously declared vars for a given session (as in Invoke-Command -Session $session).
+        /// </summary>
+        /// <param name="sessionName"></param>
+        private IEnumerable<VariableExpressionAst> GetAssignedVarsInSession(string sessionName)
+        {
+            return _varsDeclaredPerSession[sessionName];
+        }
+
+        /// <summary>
+        /// AddAssignedVarsToSession: Adds variables to the list of assigned variables for a given Invoke-Command session.
+        /// </summary>
+        /// <param name="sessionName"></param>
+        /// <param name="variablesToAdd"></param>
+        private void AddAssignedVarsToSession(string sessionName, IEnumerable<VariableExpressionAst> variablesToAdd)
+        {
+            if (!_varsDeclaredPerSession.ContainsKey(sessionName))
+            {
+                _varsDeclaredPerSession.Add(sessionName, new List<VariableExpressionAst>());
+            }
+
+            _varsDeclaredPerSession[sessionName].AddRange(variablesToAdd);
+        }
+
+        /// <summary>
+        /// AnalyzeScriptBlock: Generate a Diagnostic Record for each incorrectly used variable inside a given ScriptBlock.
+        /// </summary>
+        /// <param name="scriptBlockExpressionAst"></param>
+        private void AnalyzeScriptBlock(ScriptBlockExpressionAst scriptBlockExpressionAst)
+        {
+            var nonAssignedNonUsingVarAsts = FindNonAssignedNonUsingVarAsts(scriptBlockExpressionAst,
+                FindVarsInAssignmentAsts(scriptBlockExpressionAst)).ToList();

```

Comment:
The method below doesn't depend on the argument being a list, so this just double allocates and can be removed

```suggestion
                FindVarsInAssignmentAsts(scriptBlockExpressionAst));

```